src/FlagConfigurationLoader.php

```diff
@@ -5,56 +5,37 @@
 use Eppo\DTO\Flag;
 use Eppo\Exception\HttpRequestException;
 use Eppo\Exception\InvalidApiKeyException;
-use Psr\Http\Client\ClientExceptionInterface;
-use Psr\SimpleCache\InvalidArgumentException;
 
-class FlagConfigurationLoader
+class FlagConfigurationLoader implements IFlags
 {
     private UFCParser $parser;
-    public function __construct(private readonly APIRequestWrapper $apiRequestWrapper, private readonly ConfigurationStore $configurationStore)
-    {
+
+    public function __construct(
+        private readonly APIRequestWrapper $apiRequestWrapper,
+        private readonly IConfigurationStore $configurationStore
+    ) {
         $this->parser = new UFCParser();
     }
 
     /**
-     * @throws ClientExceptionInterface
-     * @throws InvalidArgumentException
      * @throws HttpRequestException
      * @throws InvalidApiKeyException
      */
-    public function getConfiguration(string $flagKey): ?Flag
+    public function fetchAndStoreConfigurations(): void
     {
-        if ($this->apiRequestWrapper->isUnauthorized) {
-            throw new InvalidApiKeyException();
-        }
-
-        $configuration = $this->configurationStore->getConfiguration($flagKey);
-
-        if (!$configuration) {
-            $configurations = $this->fetchAndStoreConfigurations();
-            if (!$configurations || !count($configurations) || !array_key_exists($flagKey, $configurations)) {
-                return null;
-            }
+        $responseData = json_decode($this->apiRequestWrapper->get(), true);
 
-            $configuration = $configurations[$flagKey];
+        if (!$responseData) {
+            syslog(LOG_WARNING, "[Eppo SDK] Empty or invalid response from the configuration server.");
+            return;
         }
 
-        return $this->parser->parseFlag($configuration);
+        $inflated = array_map(fn($object) => $this->parser->parseFlag($object), $responseData['flags']);
```

Comment:
Would prefer something a little more native in php for parsing from JSON string to our DTOs, but this refactor is already much better